docs/source/compression_preview/setting.rst

```diff
@@ -0,0 +1,211 @@
+        },
+        'bias': {
+            'align': {
+                'module_name': None,
+                'target_name': 'weight',
+                'dims': [0],
+            },
+            'apply_method': 'mul',
+        }
+    }
+
+We can create a setting for ``CustomizedModule`` by following the above default setting.
+
+.. code-block:: python
+
+    customized_setting = {
+        'p1': {
+            'sparse_ratio': None,
+            'max_sparse_ratio': None,
+            'min_sparse_ratio': None,
+            'sparse_threshold': None,
+            'global_group_id': None,
+            'dependency_group_id': None,
+            'granularity': [1, -1],
+            'internal_metric_block': None,
+            'apply_method': 'mul',
+        },
+        'p2': {
+            'sparse_ratio': None,
+            'max_sparse_ratio': None,
+            'min_sparse_ratio': None,
+            'sparse_threshold': None,
+            'global_group_id': None,
+            'dependency_group_id': None,
+            'granularity': [1, -1],
+            'internal_metric_block': None,
+            'apply_method': 'mul',
+        },
+        '_output_': {
+            'align': {
+                'module_name': None,
+                'target_name': 'p1',
+                'dims': [0],
+            },
+            'apply_method': 'mul',
+            'granularity': [-1, 1]
+        }
+    }
+
+    PruningSetting.register('CustomizedModule', customized_setting)
+
+The customized setting means that ``p1`` and ``p2`` will be applied channel-wise masks on the first dim of parameter,
+``_output_`` will be applied channel-wise masks on the second dim of output.
+Instead of generating masks by pruning algorithms, the output masks is generated by align with ``p1`` masks on the first dim.
+
+Quantization Setting
+^^^^^^^^^^^^^^^^^^^^
+
+Here is a default setting for quantizing module inputs, outputs and weight.
+
+.. code-block:: python
+
+    default_setting = {
+        '_input_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        'weight': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        },
+        '_output_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': 'default',
+            'apply_method': 'clamp_round',
+        }
+    }
+
+Just modified the keys and registered it to ``QuantizationSetting`` are all you need for quantizing this module.
+
+.. code-block:: python
+
+    customized_setting = {
+        '_input_': {
+            'quant_dtype': None,
+            'quant_scheme': None,
+            'granularity': [-1, 1],
+            'apply_method': 'clamp_round',
+        },
+        'p1': {
+            'quant_dtype': None,
+            'quant_scheme': None,
```

Comment:
what is the meaning of `None` for `quant_dtype` and `quant_scheme`?

Reply:
`None` means has no default value, it a template setting for this customized module type, not a config, that means when user writing a config, `quant_dtype` and `quant_scheme` should be configured.